src/main/java/org/junit/experimental/results/ResultMatchers.java

```diff
@@ -34,14 +36,21 @@ public ResultMatchers() {
      * Matches if there are {@code count} failures
      */
     public static Matcher<PrintableResult> failureCountIs(final int count) {
+        return failureCount(equalTo(count));
+    }
+
+    /**
+     * Matches if the number of failures matches {@code countMatcher}
+     */
+    public static Matcher<PrintableResult> failureCount(final Matcher<Integer> countMatcher) {
```

Comment:
The existing `failureCountIs(final int count)` only allows to do an equals match, whereas taking a Matcher of Integer allows clients to do any kind of integer matching, eg. <, >, >=, <=.
It came from the original implementation of this PR 9f5dfd8c1a9a006bc2e5d93a9bc770f2449e7afa

But since I later changed implementation (69e8a8a6b3d3d1074476463e85dcfef3b9feb0c7), it's not needed for this PR. I've removed it and I'll open a new PR for that, as I think it still makes sense on its own.

Please let me know your thoughts